### ISSUE TITLE
Set up fswatch on server

### ISSUE THREAD

**Issue by nikdragovic** [emory-libraries/templating-engine#30]
Conducted via emails to middleware.

**Comment by laurenhamel** [comment]
I'm adding this issue as blocked by #33 because we'll want to complete that test and compare its benchmarking results to #34 before making a decision in regards to how we wish to proceed.

**Comment by laurenhamel** [comment]
Based on the configuration scripts I put in place in #31, we'll need to make sure we put a `bin/` folder in place on the server to store our executables. We'll need to make sure that this `bin/` folder is private (re: not accessible to the public) and that the scripts can be executed from the folder via shell. This also means that either (a) we'll want to get shell access to configure/reconfigure our `fswatch` tasks as needed, or (b) we'll need to rely on middleware for this.

**Comment by kristianserrano** [comment]
I'll consult with Doug regarding reaching out to middleware about these requirements.

**Comment by kristianserrano** [comment]
I've sent the email to middleware. Putting this in blocked until the configuration is confirmed and tested.

**Comment by kristianserrano** [comment]
1. James has set up fswatch on both web servers.

2. To prevent both servers executing the update script, when fswatch executes on one of the servers, it’ll drop a file on the mount that indicates that it’s already indexing. The script will check for that file and not update the index if the indexing flag file exists.

3. Lauren and I will need shell access to the web server to upload any executables that we compile on our own. James has agreed to set up a directory named “fswatch” into which we can upload such files. This directory will not be publicly accessible.

~4. If Middleware does require us to switch to using something else like fswatch, Lauren has indicated that the estimate for that work would be a 1 because of how its separated.~

**Comment by kristianserrano** [comment]
Update: Middleware isn't comfortable with us having shell access to the web servers. It seems like we might not need that anyway.

* Our executables will be hosted on the mount instead of on the web servers

* James will initialize the scripts for us and configure the server to auto init them whenever the servers are rebooted.

* If we ever need to troubleshoot something with the setup, we'll need Middleware's assistance. According to James, this is something they do already.

**Comment by kristianserrano** [comment]
Moving this back to In Progress.

**Comment by kristianserrano** [comment]
This issue is blocked until Steve Collins reviews our implementation.

From Steve:
>I have had a long conversation with James on this subject.  James had to compile fswatch.  This means the software will not get updates or security fixes unless we manually do it. On a web server that is exposed to the internet and frequently scanned for vulnerabilities, it is highly preferred to have everything installed and updated via repos.  That being said, this looks like it is more of a tool than a running service.  I’ve asked James to look into how it works a little more so that we can be more comfortable with this.  Things like how fswatch will get executed and what permissions it will need.

**Comment by kristianserrano** [comment]
James followed up last night with the following:
>Here is my findings, the default monitor for FSwatch does not work with NFS mounts, but that is not a problem because you can use its "poll monitor". 
This is a simplified mode of operation, some events are not supported, please read this to get familiar with what it can and can't do. http://emcrisostomo.github.io/fswatch/doc/1.9.3/fswatch.html/Monitors.html#The-Poll-Monitor

>The only other problem is "The resource consumption of this monitor increases increases linearly with the number of files being watched (the resulting system performance will probably degrade linearly or quicker). "
So as long as you don't have hundred of thousands of files being monitored we should be fine.

>I would like to see a sample of how you are going to run it, just put it in the fswatch dir on the share when you have one ready. 
>Here is how I ran it(https://github.com/emcrisostomo/fswatch/issues/209). 


>`fswatch -mpoll_monitor -r /web/test/`
>`/web/test/HTTP_NOT_IMPLEMENTED.html.var`
>`/web/test`
>`/web/test`
>`/web/test/HTTP_PRECONDITION_FAILED.html.var`

>I was able to copy files from both machine and it alerted on the single instance I had running on libwebprod1.

>My overall conclusion is this may work, but I need confirmation from the dev's that the "poll monitor" will do what they need.

I'm going to follow up with him asking if using iWatch (http://iwatch.sourceforge.net/index.html), an option he indicated that they already support for others, solves these concerns.

**Comment by libdgg** [comment]
@kristianserrano @laurenhamel per our standup decision on 5/30 to implement the manual trigger I'm moving this issue to Sprint 15 to revisit this topic and a long term solution at a later date.

**Comment by laurenhamel** [comment]
This is no longer relevant as the templating engine is being decommissioned for Phase 2.
